src/components/utils/src/threads/thread_posix.cc

```diff
@@ -158,83 +162,90 @@ bool Thread::IsCurrentThread() const {
 
 bool Thread::start(const ThreadOptions& options) {
   LOG4CXX_AUTO_TRACE(logger_);
+  pthread_attr_t attributes;
+  int pthread_result = -1;
+  {
+    sync_primitives::AutoLock auto_lock(run_lock_);
+    // 1 - state_lock locked
```

Comment:
@SNiukalov as far as i understand, we now acquire `run_lock_` instead of `state_lock_`. If it is true, this comment section is rather deceiving. I think i'd rather remove it.

Reply:
Thanks fixed in: https://github.com/Ford-Luxoft/sdl_core/pull/34/commits/5b98c28c89c072e1baafc5ea6238058ae73c1447